### PR TITLE
Add returnDict option to UploadFile

### PR DESCRIPTION
**Pull request by woodychow**

### CODE REVIEW

**Comment by rdiankov** [comment]
let's not have return format change based on arguments. i'm ok with always returning dict (and breaking back compat) OR creating a new function where it is clear the return type is different (@ziyan )

**Comment by woodychow** [comment]
I would say let's break back compatibility then

File: python/mujincontrollerclient/controllerclientbase.py [hunk 1]
```diff
@@ -683,6 +679,19 @@ def FlushAndDownloadFile(self, filename, timeout=5):
             raise ControllerClientError(response.content.decode('utf-8'))
         return response
 
+    def FlushAndHeadFile(self, filename, timeout=5):
+        """downloads a file given filename
+
+        :return: a streaming response
+        """
+        response = self._webclient.Request('HEAD', '/file/download/', params={'filename': filename}, stream=True, timeout=timeout)
```
Comment (ziyan) [inline python/mujincontrollerclient/controllerclientbase.py]: The comment of this function is outdated.

Are you sure you need `stream=True`? I don't think you need that.

File: python/mujincontrollerclient/controllerclientbase.py [hunk 2]
```diff
@@ -613,14 +613,17 @@ def SetSceneSensorMapping(self, sensormapping, scenepk=None, usewebapi=True, tim
 
     def UploadFile(self, f, filename=None, timeout=10):
         """uploads a file managed by file handle f
+
+        Args:
+            returnDict (bool): If True, returns the response of the web call in dict format
```
Comment (ziyan) [inline python/mujincontrollerclient/controllerclientbase.py]: Comment is outdated.

File: python/mujincontrollerclient/controllerclientbase.py (hunk 1 above)
Comment (woodychow) [reply python/mujincontrollerclient/controllerclientbase.py]: That's quick. Fixed

File: python/mujincontrollerclient/controllerclientbase.py (hunk 2 above)
Comment (woodychow) [reply python/mujincontrollerclient/controllerclientbase.py]: Fixed. Thanks

File: python/mujincontrollerclient/controllerclientbase.py [hunk 3]
```diff
@@ -676,10 +679,23 @@ def FlushAndDownloadFile(self, filename, timeout=5):
...
+        :return: a dict containing "modified (datetime.datetime)" and "size (int)"
+        """
+        response = self._webclient.Request('HEAD', '/file/download/', params={'filename': filename}, timeout=timeout)
+        if response.status_code != 200:
+            raise ControllerClientError(response.content.decode('utf-8'))
+        return {
+            'modified': datetime.datetime(*email.utils.parsedate(response.headers['Last-Modified'])[:6]),
+            'size': int(response.headers['Content-Length']),
+        }
+
```
Comment (woodychow) [inline python/mujincontrollerclient/controllerclientbase.py]: @ziyan, do we still need this after the new webstack changes?

Comment (ziyan) [reply python/mujincontrollerclient/controllerclientbase.py]: Yes.

**Comment by rdiankov** [comment]
thanks~

**Pull request merged**